rosetta/failure/invalid_block.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2021 Alvalor S.A.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package failure
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type InvalidBlock struct {
+	Height  uint64
+	BlockID flow.Identifier
+	Message string
+}
+
+func (i InvalidBlock) Error() string {
+	return fmt.Sprintf("invalid block (height: %d, block: %x): %s", i.Height, i.BlockID, i.Message)
```

Comment:
This error will never be visible on the Rosetta layers; it will at most be logged internally, and in the business domain of the DPS, we refer to height and block ID, not to index and hash.

Reply:
The Rosetta API will build its own error message based on the error fields.